app/components/Form/CalculationForm.tsx

```diff
@@ -1,141 +0,0 @@
-import React from 'react';
-import {
-  FormProps,
-  IChangeEvent,
-  ErrorSchema,
-  AjvError,
-  withTheme,
-} from '@rjsf/core';
-import formTheme from '../../lib/theme/DefaultTheme';
-import { customTransformErrors } from '../../lib/theme/customTransformErrors';
-import {
-  customFormats,
-  customFormatsErrorMessages,
-} from '../../data/jsonSchemaForm/customFormats';
-
-interface CalculationSchemaFormProps<T> extends FormProps<T> {
-  onCalculate?: (formData: T, updatedField?: keyof T) => T;
-  theme?: object;
-  onSubmit: any;
-  formData: any;
-}
-interface CalculationSchemaFormState<T> {
-  originalFormData: T | undefined;
-  formData: T;
-}
-
-export default class CalculationSchemaForm<T> extends React.Component<
-  CalculationSchemaFormProps<T>,
-  CalculationSchemaFormState<T>
-> {
-  Form = withTheme(formTheme);
-  transformErrors = (errors: AjvError[]) => {
-    return customTransformErrors(errors, customFormatsErrorMessages);
-  };
-  state: CalculationSchemaFormState<T>;
-  calculating = false;
-  constructor(props: CalculationSchemaFormProps<T>) {
-    super(props);
-    const formData = this.props.formData
-      ? (this.props.formData as T)
-      : ({} as T);
-    this.state = {
-      originalFormData: formData,
-      formData,
-    };
-  }
-
-  isSavedForm = this.state?.formData
-    ? Object.keys(this.state.formData).length
-    : false;
-
-  onChange(e: IChangeEvent<T>, es?: ErrorSchema): void {
-    // Exit if we're calculating
-    if (this.calculating) {
-      return;
-    }
-
-    // Set the calculating flag to true to avoid an infinite loop
-    this.calculating = true;
-
-    // Pass through the new form data to re-render with the updated value
-    this.setState({ formData: e.formData });
-
-    // Get the updated field
-    const updatedField = this.getUpdatedField(this.state.formData, e.formData);
-
-    // Use setTimeout to call onCalculate, to let all the events fire to get everything synched up
-    setTimeout(() => this.onCalculate(e.formData, updatedField));
```

Comment:
@marcellmueller @AntBush @VB-at-Bis  When you're looking for reusable code for a problem or reviewing a PR, this kind of code should be considered a red flag. Using `setTimeout` or `useEffect` to work around issues where the state isn't up to date will almost always result in race conditions and bugs. In this case, some form updates where being skipped (if you typed fast enough characters were skipped) and the calculations were sometimes done using the previous form values, resulting in incorrect totals on the review page.